utils/utils.go

```diff
@@ -601,3 +602,30 @@ func ExtractSha256FromResponseBody(body []byte) (string, error) {
 func Pointer[K any](val K) *K {
 	return &val
 }
+
+func SetEnvWithResetCallback(key, value string) (func() error, error) {
+	oldValue, exist := os.LookupEnv(key)
+	errMsg := "failed %s %s as environment variable. Cause: %s"
+
+	if err := os.Setenv(key, value); err != nil {
+		log.Debug(fmt.Sprintf(errMsg, "setting", key, err.Error()))
```

Comment:
There's no need for a debug log message, to signal that an error has occurred. It is sufficient to return this error. If you think it's important to add additional context to the error, you can wrap the error with a new error, to and add the additional information to the new error.
This also applies to the below lines.